### PR TITLE
http: Port Inbound to new data model

### PR DESCRIPTION
**Pull request by abhinav**
This ports the HTTP inbound over to the new data model.

### CODE REVIEW

**Comment by abhinav** [comment]
Ready for eyes

File: yarpc-core/src/main/java/io/yarpc/core/transport/InvalidTimeoutException.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        super(format(service, procedure, timeout));
+    }
+
+    public InvalidTimeoutException(String service, String procedure, String timeout, Throwable cause) {
+        super(format(service, procedure, timeout), cause);
+    }
+
+    private static String format(String service, String procedure, String timeout) {
+        return String.format(
+            "invalid TTL \"%s\" for procedure \"%s\" of service \"%s\": must be a positive integer",
```
Comment (breerly) [inline yarpc-core/src/main/java/io/yarpc/core/transport/InvalidTimeoutException.java]: Nice, away with the constant glossary

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInbound.java [hunk 2]
```diff
@@ -21,55 +21,139 @@
...
     }
 
-    public void setHandlerRegistry(HandlerRegistry handlerRegistry) {
-        this.handlerRegistry = handlerRegistry;
+    @Override
+    public void start() {
+        lifecycle.start(this::realStart);
+    }
+
+    private void realStart() {
```
Comment (breerly) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInbound.java]: This name seems odd, what exactly does this method do?

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInbound.java [hunk 3]
```diff
@@ -21,55 +21,139 @@
...
     @Override
-    public void stop() throws InterruptedException {
-        this.serverChannelFuture.channel().close().awaitUninterruptibly().sync();
-        this.workerGroup.shutdownGracefully().sync();
-        this.bossGroup.shutdownGracefully().sync();
+    public void stop() {
+        lifecycle.stop(this::realStop);
+    }
+
+    private void realStop() {
```
Comment (breerly) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInbound.java]: Yeah... let's figure out a better name for these.

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInboundRequestHandler.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+        try {
+            return Duration.ofMillis(Long.parseLong(timeout));
+        } catch (NumberFormatException e) {
+            throw new InvalidTimeoutException(service, procedure, timeout, e);
+        }
+    }
+
+    @Override
+    protected void channelRead0(ChannelHandlerContext ctx, FullHttpRequest httpRequest) throws Exception {
```
Comment (breerly) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInboundRequestHandler.java]: I usually expect method names to start with a verb in 'ze Java

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/package-info.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+/**
+ * Provides a YARPC transport that sends and receives requests over HTTP/1.1.
+ * <p>
+ * The transport provides first-class support for Unary RPCs and experimental support for Oneway RPCs.
+ */
+package io.yarpc.transport.http;
```
Comment (breerly) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/package-info.java]: good package name 👍

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInbound.java (hunk 2 above)
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInbound.java]: Renamed

File: yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInboundRequestHandler.java (hunk 4 above)
Comment (abhinav) [inline yarpc-transport-http/src/main/java/io/yarpc/transport/http/HTTPInboundRequestHandler.java]: Yeah, Netty didn't choose the best name here. This method has been renamed in Netty 5 but we're not using that yet.

**Pull request merged**